File: src/org/cappuccino_project/ide/intellij/plugin/decompiler/ObjJSJFileType.java
package org.cappuccino_project.ide.intellij.plugin.decompiler;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.cappuccino_project.ide.intellij.plugin.lang.ObjJIcons;
import org.cappuccino_project.ide.intellij.plugin.lang.ObjJLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class ObjJSJFileType extends LanguageFileType {
    public static final String FILE_EXTENSION = "sj";
    public static final ObjJSJFileType INSTANCE = new ObjJSJFileType();

    private ObjJSJFileType() {
        super(ObjJLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Objective-J Compiled Source";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Compiled Objective-J source code";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return FILE_EXTENSION;
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return ObjJIcons.SJ_DOCUMENT_ICON;
    }
}

File: src/cappuccino/ide/intellij/plugin/decompiler/lang/ObjJSjFileType.kt
package cappuccino.ide.intellij.plugin.decompiler.lang;

import cappuccino.ide.intellij.plugin.lang.ObjJIcons
import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.vfs.VirtualFile
import javax.swing.Icon

class ObjJSjFileType private constructor() : FileType {
        override fun isBinary(): Boolean = true

        override fun isReadOnly(): Boolean = true

        override fun getCharset(p0: VirtualFile, p1: ByteArray): String? = null

        override fun getName(): String =
                "Objective-J Compiled Script"

        override fun getDescription(): String =
                "An Objective-J compiled script file for the Cappuccino Web Framework"

        override fun getDefaultExtension(): String =
                FILE_EXTENSION

        override fun getIcon(): Icon? =
                ObjJIcons.DOCUMENT_ICON

        companion object {
                const val FILE_EXTENSION = "sj"
                val INSTANCE = ObjJSjFileType()
        }
}
